app/src/main/java/com/omkarmoghe/pokemap/MainActivity.java

```diff
@@ -14,35 +13,39 @@
 
 import com.google.android.gms.maps.GoogleMap;
 import com.google.android.gms.maps.OnMapReadyCallback;
+import com.omkarmoghe.pokemap.app.App;
 import com.omkarmoghe.pokemap.common.BaseActivity;
 import com.omkarmoghe.pokemap.map.MapWrapperFragment;
 import com.omkarmoghe.pokemap.settings.SettingsActivity;
 
-public class MainActivity extends BaseActivity implements OnMapReadyCallback{
+import javax.inject.Inject;
+
+public class MainActivity extends BaseActivity implements OnMapReadyCallback {
 
     public static final String TAG = "Pokemap";
 
     // fragments
     private MapWrapperFragment mMapWrapperFragment;
 
     // Preferences
+    @Inject
     SharedPreferences pref;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        pref = PreferenceManager.getDefaultSharedPreferences(this);
+        ((App) getApplication()).getMainComponent().inject(this);
```

Comment:
getMainComponent() should be a static accessor (to a static field). That way you can access it without casting getApplication().